tutorials/rag/evaluate-rag-flow.ipynb

```diff
@@ -842,6 +832,277 @@
     "\n",
     "![Eval Logs table](../../assets/images/evaluate_rag_flow_logs.png)"
    ]
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "# Using Retriever as a Tool\n",
+    "\n",
+    "In the above example, the flow of our app is static - given the question we always call the retriever, pass the context to the model and then respond to the user. It can be helpful to have a more dynamic setup where we instead allow the model to choose when use the retriever.\n",
+    "This allows the model to do smart query expansion, or ask for clarifying questions, before deciding to retrieve the context which can be significant boost in performance.\n",
+    "\n",
+    "This entails updating the Prompt with a Tool schema definition for the retriever. And then routing your flow to the retriever based on when or not the model makes a tool call. The Flow on Humanloop can capture the full trace of events so that you have full transparency into the models actions. And the `evaluations.run(...)` utility works as normal. We will demonstrate how to do this below.\n",
```

Comment:
"And then routing your flow to the retriever based on when or not the model makes a tool call" isn't a proper sentence. (also it should be "whether or not", which can be shortened to "whether")

Reply:
This needs to be all deleted

Reply:
standby